desc/objectives/_free_boundary.py

```diff
@@ -901,6 +871,369 @@ def _print(fmt, fmax, fmin, fmean, f0max, f0min, f0mean, norm, unit):
+        )
+        errorif(
+            not self._is_neumann and field.N_Phi_coil > grid.N,
+            msg=f"N_Phi_coil = {getattr(field, 'N_Phi_coil', 0)} > {grid.N} = grid.N.",
+        )
+        eval_grid = setdefault(eval_grid, grid)
+
+        self._field = field
+        self._B_coil = field._B0 if self._is_neumann else field._B_coil
+        self._grid = grid
+        self._eval_grid = eval_grid
+        self._coil_grid = coil_grid
+        self._use_same_grid = grid.equiv(eval_grid)
+        self._q = q
+        self._maxiter = maxiter
+        self._chunk_size = chunk_size
+        self._B_coil_chunk_size = B_coil_chunk_size
+        self._grad_keys = ["grad(theta)", "grad(zeta)", "n_rho"]
+        self._inner_keys = ["|B|^2", "p", "I", "|e_theta x e_zeta|"] + self._grad_keys
+        self._reuseable_keys = [
+            "0",
+            "R",
+            "R_t",
+            "R_z",
+            "Z_t",
+            "Z_z",
+            "e_theta",
+            "e_theta x e_zeta",
+            "e_zeta",
+            "n_rho",
+            "omega_t",
+            "omega_z",
+            "|e_theta x e_zeta|",
+        ]
+        self._I_sheet = I_sheet
+
+        super().__init__(
+            things=eq,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+            jac_chunk_size=jac_chunk_size,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self.things[0]
+
+        # TODO (#1243, #1154): Partial summation to evaluate Rb_lmn only.
+        #   Then pass in transforms=eq_transforms in eval_transforms
+        eq_transforms = get_transforms(self._inner_keys, eq, grid=self._eval_grid)
+        eval_transforms = get_transforms("|K_vc|^2", self._field, grid=self._eval_grid)
+        if self._use_same_grid:
+            source_transforms = eval_transforms
+            grad_transforms = eq_transforms
+        else:
+            source_transforms = get_transforms("Phi_mn", self._field, grid=self._grid)
+            grad_transforms = get_transforms(self._grad_keys, eq, grid=self._grid)
+
+        data, _ = self._field.compute(
+            ["interpolator"] if self._is_neumann else ["interpolator", "Y_coil"],
+            grid=self._grid,
+            q=self._q,
+            transforms=source_transforms,
+            B_coil=self._B_coil,
+            B_coil_chunk_size=self._B_coil_chunk_size,
+        )
+        # No net poloidal current in equation 4.13 of [1].
+        self._field.Y = 0.0 if self._is_neumann else data["Y_coil"]
```

Comment:
As `obj.build` is not jitted, this line should execute and permanently modify self._field.Y.